acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/privacy/unrestricted/DeployUnrestrictedSmartContractAcceptanceTest.java

```diff
@@ -0,0 +1,58 @@
+/*
```

Comment:
Agreed, there's definitely a better way of doing these tests. We should be able to fire in a matrix of combinations  `unrestricted` `restricted` `onchain` etc and come out with some reasonable results depending on how it's setup.

Reply:
Something like https://github.com/hyperledger/besu/pull/2324/commits/b2f43f5f4d72ddddd68f1e15f78e12608d6cb9c5 perhaps? Feels a little clunky but removes the duplication around asserting the same thing happens in different privacy modes. It would be nice to get to the point where the `PrivacyNode` and the `Restriction` is the parameters but would require too much of a jump/refactor of existing tests.

Reply:
Had to squash rebase from master as there were too many changes and conflicts. This is what I ended up with in the end to solve this.

https://github.com/antonydenyer/besu/blob/unrestricted-privacy-endpoints/acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/privacy/ParameterizedEnclaveTestBase.java#L45

Reply:
The only problem with this approach is that some of the tests are moot. So you end up with a couple of things like https://github.com/antonydenyer/besu/blob/unrestricted-privacy-endpoints/acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/privacy/PrivDebugGetStateRootOffchainGroupAcceptanceTest.java#L91 for when it doesn't make sense. 

Ideally you'd want an annotation to be able to do something like `@Test(exclude= () -> { restriction == UNRESTRICTED)})` but I don't know of a nice way of doing that in java?